tests/unit/s2n_config_test.c

```diff
@@ -255,23 +255,6 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_connection_free(conn));
             EXPECT_SUCCESS(s2n_config_free(config));
         };
-
-        /* Test that security policy validation is enforced on the config */
-        {
-            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
-            EXPECT_NOT_NULL(config);
-            DEFER_CLEANUP(struct s2n_connection *conn = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
-            EXPECT_NOT_NULL(conn);
-
-            DEFER_CLEANUP(struct s2n_cert_chain_and_key *invalid_cert = NULL, s2n_cert_chain_and_key_ptr_free);
-            EXPECT_SUCCESS(s2n_test_cert_permutation_load_server_chain(&invalid_cert, "rsae", "pss", "4096", "sha384"));
-            EXPECT_SUCCESS(s2n_config_add_cert_chain_and_key_to_store(config, invalid_cert));
-            struct s2n_security_policy rfc9151_applied_locally = security_policy_rfc9151;
-            rfc9151_applied_locally.certificate_preferences_apply_locally = true;
-            config->security_policy = &rfc9151_applied_locally;
-
-            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_set_config(conn, config), S2N_ERR_SECURITY_POLICY_INCOMPATIBLE_CERT);
-        };
```

Comment:
Do you have another test where s2n_connection_set_config fails due to a security policy override + S2N_ERR_SECURITY_POLICY_INCOMPATIBLE_CERT?

Maybe you'd also want a test where you manually construct an config with certs not valid for the security policy and confirm that s2n_connection_set_config doesn't reject it. That would give confidence that s2n_connection_set_config isn't doing the expensive work of validating the config.

Reply:
Yes, that case is covered here: https://github.com/aws/s2n-tls/blob/9371c90f3852db19429875e2d9a553d87aff9301/tests/unit/s2n_connection_test.c#L630-L635

And that sounds like a good idea! For context this was actually validated using the new regression tooling that we were working on, and we were able to show that the instruction counts for `s2n_connection_set_config` decreased.